system/security/LdapSecurity/ldapconnection.cpp

```diff
@@ -2642,7 +2650,17 @@ class CLdapClient : implements ILdapClient, public CInterface
                 dispname_values
             };
 
-            LDAPMod *attrs[4];
+
+            StringBuffer emplID(user.getEmployeeID());
```

Comment:
I think this would be ugly because because both employeeID_values and employeeID_attr structures would still have to be defined outside the isEmpty() test, and then conditionally set within.  The pattern throughout the rest of this file is to skip the isEmpty test, so unless you feel strongly I prefer to leave it as is. As a consolation, I do test that the employeeID is not NULL before adding to the attribute array (line 2678)